rule/add_constant.go

```diff
@@ -42,7 +42,11 @@ type AddConstantRule struct {
 
 // Apply applies the rule to given file.
 func (r *AddConstantRule) Apply(file *lint.File, arguments lint.Arguments) []lint.Failure {
-	r.configureOnce.Do(func() { r.configure(arguments) })
+	var configureErr error
+	r.configureOnce.Do(func() { configureErr = r.configure(arguments) })
+	if configureErr != nil {
+		return []lint.Failure{lint.NewInternalFailure(configureErr.Error())}
+	}
```

Comment:
This idea is great, but it doesn't address this

https://github.com/mgechev/revive/pull/1126#discussion_r1874704591

But it could be addressed separately

Reply:
Yes, I know. It is not my intention to address this problem right now.
IMO configureOnce is cumbersome (vs just lock/unlock) for dealing with errors

Reply:
Yes, I agree

Maybe linter could call configure if there is a Configure  interface on the Rule

But, some tests will need to be rewritten

Reply:
I've studied before the alternative of adding a `Configure` function to rules: it simplifies configuration of rules but it has the drawback of changing the interface.

Reply:
My point was to add an interface for Configure only.

type Configure interface {
   Configure() error
}

If detected, you call it after a type assertion.

But it's maybe part of what you considered

Reply:
ok @chavacava nice concept, can you deploy that changes for lint/* files without add_constant.go rule, and in #1126 do the rest of job (add ``` []lint.Failure{lint.NewInternalFailure(configureErr.Error())} ``` where it is needed) :P